### PR TITLE
Obmin 266 checkbox

### PR DESCRIPTION
**Pull request by AkintsevaElizaveta**
Created checkbox component

### CODE REVIEW

File: src/components/checkbox/styles.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import styled, { css } from 'styled-components';
+import { DivArg, LabelArg, LabelSquareArg } from './types';
```
Comment (Mish-hash) [inline src/components/checkbox/styles.ts:2]: missing empty line between different import type

File: src/components/checkbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+const CheckBox = ({
+                    gap,
+                    text,
+                    margin,
+                    onClick,
+                    fontSize,
+                    checked = false,
+                  }: CheckboxProps) => (
```
Comment (Wolshebnik) [inline src/components/checkbox/index.tsx:14]: We need to configure the formatting
If you use webstorm, it has its own formatting

File: src/components/checkbox/Checkbox.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  checked: false,
+};
+
+export const CheckboxChecked = Template.bind({});
+CheckboxChecked.args = {
+  checked: true,
+  margin: 26,
+  text: 'checkedCheckbox',
+  fontSize: 18,
+};
```
Comment (Wolshebnik) [inline src/components/checkbox/Checkbox.stories.tsx:28]: add an empty line to the end of the file

File: src/components/checkbox/styles.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  `}
+  & > svg {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+  }
+
+  &:hover {
+  }
```
Comment (Wolshebnik) [inline src/components/checkbox/styles.ts:26]: If hover is empty
Remove it

File: src/components/checkbox/styles.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  `}
+`;
+
+const Div = styled.div<DivArg>`
+  display: flex;
+  transition: all ease-in-out 0.3s;
+  margin: 0;
+
+  ${(p) => css`
+    margin: ${p.margin || 22 + 'px'};
```
Comment (Wolshebnik) [inline src/components/checkbox/styles.ts:49]: ${(p) => css``}
rewrite
${({theme, checked}) => css``}

File: src/components/checkbox/types.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface LabelArg {
+  gap?: number;
+  checked: boolean;
+  fontSize: number;
+}
+
+export interface LabelSquareArg {
+  checked: boolean;
+}
```
Comment (Wolshebnik) [inline src/components/checkbox/types.ts:23]: add an empty line to the end of the file

File: src/components/checkbox/styles.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+const Label = styled.label<LabelArg>`
+  display: block;
+  line-height: 17px;
+  cursor: pointer;
+  max-width: 679px;
+  width: 100%;
+
+  ${({ theme: { colors }, gap, checked, fontSize }) => css`
+    margin-left: ${(gap || 22 + 'px')};
+    font-size: ${fontSize || 14 + 'px'};
```
Comment (Wolshebnik) [inline src/components/checkbox/styles.ts:38]: rewrite pls
margin-left: ${gap || 22}px;
font-size: ${fontSize || 14}px;

File: src/components/checkbox/styles.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+`;
+
+const Label = styled.label<LabelArg>`
+  display: block;
+  line-height: 17px;
+  cursor: pointer;
+  max-width: 679px;
+  width: 100%;
+
+  ${({ theme: { colors }, gap, checked, fontSize }) => css`
```
Comment (Wolshebnik) [inline src/components/checkbox/styles.ts:36]: remove colors
I wrote an example above

File: src/components/checkbox/styles.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  box-sizing: border-box;
+  width: 17px;
+  height: 17px;
+  vertical-align: bottom;
+  cursor: pointer;
+  transition: all ease-in-out 0.3s;
+
+  ${({ theme: { colors }, checked }) => css`
+    background-color: ${checked && colors.btnBlue};
+    border: 3px solid ${checked ? colors.btnBlue : colors.colorGrey};
```
Comment (Wolshebnik) [inline src/components/checkbox/styles.ts:16]: rewrite pls
${({ theme, checked }) => css`
    background-color: ${checked && theme.colors.btnBlue};
    border: 3px solid ${checked ? theme.colors.btnBlue : theme.colors.colorGrey};
  `}

File: src/components/checkbox/index.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  <Div
+    margin={margin}
+    onClick={onClick}
+    checked={checked}
+  >
+    <LabelSquare checked={checked}>
+      <CheckSvg />
+    </LabelSquare>
+
+    <div>
```
Comment (Wolshebnik) [inline src/components/checkbox/index.tsx:24]: Are you sure this div is needed?

**Pull request merged**